### PR TITLE
fix status issue

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 1]
```diff
@@ -1190,40 +1191,59 @@ public static String retrieveDeployedAPI(String apiName, String version, String
...
-    public static API getAPI(org.apache.synapse.MessageContext messageContext) {
+    public static boolean isAPIStatusProtoType(org.apache.synapse.MessageContext messageContext) {
+
+        return APIConstants.PROTOTYPED.equals(getStatus(messageContext));
+    }
+
+    public static String getAPINameFromContextAndVersion(org.apache.synapse.MessageContext messageContext) {
+
+        API api = getAPI(messageContext);
+        if (api != null){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:1237]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 2]
```diff
@@ -1190,40 +1191,59 @@ public static String retrieveDeployedAPI(String apiName, String version, String
...
 
-        String context = (String) messageContext.getProperty(RESTConstants.REST_API_CONTEXT);
-        String version = (String) messageContext.getProperty(RESTConstants.SYNAPSE_REST_API_VERSION);
-        SubscriptionDataStore tenantSubscriptionStore =
-                SubscriptionDataHolder.getInstance().getTenantSubscriptionStore(getTenantDomain());
-        if (tenantSubscriptionStore != null) {
-            return tenantSubscriptionStore.getApiByContextAndVersion(context, version);
+    public static String getApiProviderFromContextAndVersion(org.apache.synapse.MessageContext messageContext) {
+        API api = getAPI(messageContext);
+        if (api != null){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:1245]: Formatting issue

**Review by mushthaq33 — APPROVED** [review]

**Comment by tharindu1st** [comment]
integration test change https://github.com/wso2/product-apim/pull/9984

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtGoogleAnalyticsTrackingHandler.java [hunk 3]
```diff
@@ -71,6 +72,9 @@
     @MethodStats
     @Override
 	public boolean handleRequest(MessageContext msgCtx) {
+        if (GatewayUtils.isAPIStatusProtoType(msgCtx)) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtGoogleAnalyticsTrackingHandler.java:75]:
```suggestion
        if (GatewayUtils.isAPIStatusPrototype(msgCtx)) {
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 4]
```diff
@@ -4087,6 +4088,40 @@ paths:
...
+    post:
+      tags:
+        - APIs
+      summary: Generate internal API Key to invoke APIS.
+      description: |
+        This operation can be used to generate internal api key which used to invoke API.
+      operationId: generateInternalAPIKey
+      parameters:
+        - $ref: '#/components/parameters/apiId'
+        - $ref: '#/components/parameters/If-None-Match'
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:4101]: remove if not supported

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 5]
```diff
@@ -4087,6 +4088,40 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/APIKey'
+        400:
+          $ref: '#/components/responses/BadRequest'
+        404:
+          $ref: '#/components/responses/NotFound'
+        412:
+          $ref: '#/components/responses/PreconditionFailed'
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:4116]: remove if not supported

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 6]
```diff
@@ -4087,6 +4088,40 @@ paths:
...
+      security:
+        - OAuth2Security:
+            - apim:api_create
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:4119]: shall we use a new scope for this like : apim:api_generate_key and assign the relevant roles

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**